### DISCUSSION TITLE
Is it possible to access a raw message? If not, why?

### DISCUSSION THREAD

**Discussion by UnFefeSauvage** [Rapptz/discord.py#6303] [Q&A] [answered]
~~I wanted to access the message answered by another (the new Discord feature)  and I realized it wasn't handled by discord.py yet.~~
**It is, in fact accessible under [Message.reference](https://discordpy.readthedocs.io/en/latest/api.html#discord.Message.reference), thank you Rapptz.**
It is handled by the Discord API though so I tried to access the raw message to do it manually: I didn't find any way to access it.
So: is there a way to access the raw version of a message?
I've seen several *"raw"* fields in the message class, why isn't there the full raw message?

**Comment by Rapptz** [comment]
I'm not really sure what this is asking. What would a 'raw' message entail? The Discord API has no such concept -- it is a library only concept that denotes it's a lower level API.

**Reply by UnFefeSauvage** [reply]
What i meant by a 'raw' message is the equivalent of what returned the lower level API as a python `dict` (in the same way as the payloads given to the `on_raw_message_edit` and `on_raw_message_delete`).
In my specific case, I am trying to access the `referenced_message` field from the [message object.](https://discord.com/developers/docs/resources/channel#message-object-message-structure)

Thank you for your time!

**Reply by Rapptz** [reply]
No there is no such thing, nor will there ever be. 

If you want to access `referenced_message` then it's under [`Message.reference`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Message.reference).

**Reply by UnFefeSauvage** [reply]
My bad, I didn't look close enough this field... Thanks a lot!

**Comment by Werseter** [comment, marked as answer]
Wouldn't `on_socket_raw_receive` contain the aforementioned JSON (parsable to dict)? It might be an overkill and not be exactly practical way to access the referenced ID, but it is not exactly impossible, right?

**Reply by UnFefeSauvage** [reply]
Rapptz solved my problem but that's the answer to the question indeed!

**Reply by Rapptz** [reply]
No, `on_socket_raw_receive` would give you raw bytes not a dict.

**Comment by Werseter** [comment]
Oh my mails have not fetched yet and I didn't see the appropriate, already existing reply. But still, regarding the dict, it is still something to perhaps have stashed somewhere it the back of your mind.

**Comment by Werseter** [comment]
Yes, of a parsable JSON? Am I remembering wrongly? Please do correct me if I'm wrong, I believe I did mention it offering a JSON, not a dict per se.

**Reply by Rapptz** [reply]
No. It doesn't give you JSON at all. It gives you binary frames. You're supposed to handle these yourself by making the buffer and unzipping them etc. It's not meant to be used to obtain JSON because it just duplicates the library's parsing.

**Comment by Werseter** [comment]
Alright then, perhaps I did remember it wrongly then. I do recall printing out the parsed results from it a few years back though, and I've thought it was in a JSON format. I am sorry for causing some confusion here.
